src/nat/builder/sync_builder.py

```
@@ -0,0 +1,571 @@
+    Provides synchronous access to Builder methods by wrapping async calls with run_until_complete.
+    """
+
+    def __init__(self, builder: Builder) -> None:
+        self._builder = builder
+
+        try:
+            # Save the current loop. This should always be available given the creation pattern of the Builder class.
+            self._loop = asyncio.get_running_loop()
+        except RuntimeError as e:
+            raise ValueError("No event loop is running. If you are running the code in a synchronous context, "
+                             "please use the async builder instead.") from e
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)